src/common/CustomSignalHandler.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+    SPDX-FileCopyrightText: 2021 Aleix Pol Gonzalez <aleixpol@kde.org>
+
+    SPDX-License-Identifier: LGPL-2.0-only
+*/
+
+#include "CustomSignalHandler.h"
+#include <QDebug>
+#include <signal.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+int CustomSignalHandler::signalFd[2];
+
+CustomSignalHandler::CustomSignalHandler()
```

Comment:
We also have src/daemon/SignalHandler.cpp:

which is near identical to this class. Only more verbose


We don't want both.
 
Can we either port:
src/daemon/DaemonApp.cpp:        connect(m_signalHandler, &SignalHandler::sigintReceived, this, &DaemonApp::quit);
src/daemon/DaemonApp.cpp:        connect(m_signalHandler, &SignalHandler::sigtermReceived, this, &DaemonApp::quit);

to this and drop SignalHandler.h or vice-versa.

Reply:
Fair enough, added custom signal handling on the previous one so that we changed the least and moved it to common.

This class could use a cleanup, there's quite a bit of unneeded duplicate code as far as I understand, probably not for this MR though.